modules/dxp/apps/commerce-avalara/commerce-avalara-tax-engine-fixed-web/src/main/java/com/liferay/commerce/avalara/tax/engine/fixed/web/internal/portlet/action/EditCommerceTaxAvalaraMVCActionCommand.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ *
+ *
+ *
+ */
+
+package com.liferay.commerce.avalara.tax.engine.fixed.web.internal.portlet.action;
+
+import com.liferay.commerce.avalara.tax.engine.fixed.configuration.CommerceTaxAvalaraTypeConfiguration;
+import com.liferay.commerce.constants.CommercePortletKeys;
+import com.liferay.commerce.tax.model.CommerceTaxMethod;
+import com.liferay.commerce.tax.service.CommerceTaxMethodService;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCActionCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCActionCommand;
+import com.liferay.portal.kernel.settings.GroupServiceSettingsLocator;
+import com.liferay.portal.kernel.settings.ModifiableSettings;
+import com.liferay.portal.kernel.settings.Settings;
+import com.liferay.portal.kernel.settings.SettingsFactory;
+import com.liferay.portal.kernel.util.Constants;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Portal;
+
+import javax.portlet.ActionRequest;
+import javax.portlet.ActionResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Calvin Keum
+ */
+@Component(
+	enabled = true, immediate = true,
+	property = {
+		"javax.portlet.name=" + CommercePortletKeys.COMMERCE_TAX_METHODS,
+		"mvc.command.name=editCommerceAvalaraConnector"
+	},
+	service = MVCActionCommand.class
+)
+public class EditCommerceTaxAvalaraMVCActionCommand
+	extends BaseMVCActionCommand {
+
+	@Override
+	protected void doProcessAction(
+			ActionRequest actionRequest, ActionResponse actionResponse)
+		throws Exception {
+
+		String cmd = ParamUtil.getString(actionRequest, Constants.CMD);
+
+		if (cmd.equals(Constants.UPDATE)) {
+			_updateCommerceTaxAvalara(actionRequest);
+		}
+	}
+
+	private void _updateCommerceTaxAvalara(ActionRequest actionRequest)
+		throws Exception {
+
+		long commerceTaxMethodId = ParamUtil.getLong(
+			actionRequest, "commerceTaxMethodId");
+
+		CommerceTaxMethod commerceTaxMethod =
+			_commerceTaxMethodService.getCommerceTaxMethod(commerceTaxMethodId);
+
+		Settings settings = _settingsFactory.getSettings(
+			new GroupServiceSettingsLocator(
+				commerceTaxMethod.getGroupId(),
+				CommerceTaxAvalaraTypeConfiguration.class.getName()));
+
+		ModifiableSettings modifiableSettings =
+			settings.getModifiableSettings();
+
+		String accountNumber = ParamUtil.getString(
```

Comment:
@dsitu should we group all param definition first and than use them in the objct setters?

Reply:
Hey @marco-leo good question, this is actually ok to do in Liferay. I believe it follows the order by usage pattern. I know that in Commerce we have a exception that if there are usually a lot of params we do group them together then set them. I noticed that other Commerce files that are setting `modifiableSettings` are not grouping the params either. These, `EditPayPalCommercePaymentMethodConfigurationMVCActionCommand`, `EditAuthorizeNetCommercePaymentMethodConfigurationMVCActionCommand` and `EditMercanetCommercePaymentMethodConfigurationMVCActionCommand`. I think we should follow the same format for all of them. So in this case we can leave them ungrouped. But if you prefer we group them, we can do that too. I can apply the SF change to those three files. Let me know what you think.